#!/usr/bin/env python
PACKAGE = "ardrone_velocity_ekf"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp",     double_t,    0, "Proportional Coefficient in X direction", 0.15,  0.0, 1.0)
gen.add("Ki",     double_t,    0, "Integral Coefficient in X direction", 0.07,  0.0, 0.5)
gen.add("Kd",     double_t,    0, "Derivative Coefficient in X direction", 0.07,  0.0, 0.5)
gen.add("Kp_yaw",     double_t,    0, "Proportional Coefficient for yaw angle", 0.0,  0.0, 1.0)
gen.add("limit_x",     double_t,    0, "Maximum control output in x direction", 0.5,  0.0, 1.0)
gen.add("limit_y",     double_t,    0, "Maximum control output in y direction", 0.5,  0.0, 1.0)
gen.add("windup",     double_t,    0, "Windup  Coefficient of Integral Term", 0.6,  0.0, 1.0)
gen.add("beta",     double_t,    0, "Set point weighting", 1.0,  0.0, 1.0)
gen.add("derv_filter",     double_t,    0, "Derivative Low-Pass", 0.0,  0.0, 1.0)
gen.add("derv_median",     double_t,    0, "Derivative 3-Point Median", 0.0,  0.0, 1.0)
gen.add("derv_smith",     double_t,    0, "Derivative Smith", 1.0,  0.0, 1.0)
gen.add("derv_error",     double_t,    0, "Derivative Error", 0.0,  0.0, 1.0)
gen.add("derv_vel",     double_t,    0, "Derivative Velocity", 1.0,  0.0, 1.0)
exit(gen.generate(PACKAGE, "velocity_control", "dynamic_param_config"))
